test/test_torch.py

```diff
@@ -14639,6 +14639,40 @@ def test_ctor_with_numpy_array(self, device):
             for i in range(len(array)):
                 self.assertEqual(tensor[i], array[i])
 
+    @unittest.skipIf(not TEST_NUMPY, "Numpy not found")
+    @dtypes(*torch.testing.get_all_dtypes())
+    def test_numpy_scalar_cmp(self, device, dtype):
+        t = torch.tensor([0, 1, 2], dtype=dtype, device=device)
+        if dtype == torch.bfloat16:
+            with self.assertRaises(TypeError):
+                np_array = t.cpu().numpy()
+            return
+
+        np_array = t.cpu().numpy()
+        self.assertEqual(t[0], np_array[0])
+        self.assertEqual(t[0].item(), np_array[0].item())
+        self.assertEqual(t[0], np_array[0].item())
+        self.assertEqual(t[0].item(), np_array[0])
+        self.assertTrue(t[0] == np_array[0])
+        self.assertTrue(t[0].item() == np_array[0].item())
+        self.assertTrue(t[0] == np_array[0].item())
+        self.assertTrue(t[0].item() == np_array[0])
+
+        if dtype.is_complex:
+            t = torch.tensor([complex(1, 2), complex(0, 3)], dtype=dtype, device=device)
+            np_array = t.cpu().numpy()
+            self.assertEqual(t[0], np_array[0])
+            self.assertEqual(t[0].item(), np_array[0].item())
+            self.assertEqual(t[0], np_array[0].item())
+            self.assertEqual(t[0].item(), np_array[0])
+            self.assertTrue(t[0].item() == np_array[0].item())
+            self.assertTrue(t[0] == np_array[0].item())
+            self.assertTrue(t[0].item() == np_array[0])
+            if dtype == torch.complex64:  # TODO: Imaginary part is dropped in this case. Need fix.
```

Comment:
Interesting... so this works for complex128? That's really weird.

Reply:
Yes. I'm not sure what the cause is...

Reply:
That's really interesting. Would you cite the linked issue in this comment, too, so people know where to look for the status of this problem?